src/js/bg/plugins/calls/call/sip.js

```diff
@@ -180,27 +190,34 @@ class CallSIP extends Call {
             this.session.bye()
         })
 
-        this.session.on('failed', (message) => {
-            this.app.logger.info(`${this}call declined: ${message.status_code}/${this.state.stats.callId}`)
+        this.session.on('failed', (response) => {
             this.busyTone.play()
 
-            if (message.status_code === 480) {
-                // Temporarily Unavailable; Callee currently unavailable.
-                this.setState({status: 'callee_unavailable'})
-            } else if (message.status_code === 486) {
-                // Busy here; Callee is busy.
-                this.setState({status: 'callee_busy'})
-            } else if (message.status_code === 487) {
-                // Request terminated; Request has terminated by bye or cancel.
-                this.setState({status: 'request_terminated'})
+            // See https://sipjs.com/api/0.11.0/session/#failed
+            // The response parameter should be a `SIP.IncomingResponse`. 
+            // The type is not consistent though. It can also be a SIP header 
+            // string or just null.     
+            if (response) {
+                this.app.logger.info(`${this}call declined: ${response.status_code}/${this.state.stats.callId}`)
+                if (response.status_code === 480) {
```

Comment:
Can't we use `response.cause` and match it against the constants in `SIP.C.causes`, for example:
`SIP.C.causes.REJECTED`? (like it's done in the example [here](https://sipjs.com/api/0.11.0/causes/))

Reply:
Good idea. I'll update the pull request with these constants.